packages/core-transactions/src/errors.ts

```diff
@@ -90,6 +90,15 @@ export class WalletUsernameAlreadyRegisteredError extends TransactionError {
     }
 }
 
+export class WalletUsernameMismatchError extends TransactionError {
```

Comment:
The resignation can only be made by delegates. Therefore we don't need a username.